src/workerd/server/server.c++

```diff
@@ -2391,14 +2456,173 @@ kj::Promise<void> Server::run(jsg::V8System& v8System, config::Config::Reader co
         "override provided on the command line."));
   }
 
-  // We should have registered all headers synchronously. This is important becaues we want to
-  // be able to start handling requests as soon as the services are available, even if some other
-  // services take longer to get ready.
+  return tasks.onEmpty();
+}
+
+// =======================================================================================
+// Server::test()
+
+namespace {
+
+class GlobFilter {
+  // Implements glob filters. Copied from kj/test.{h,c++}, modified only to avoid copying the
+  // pattern in the constructor.
+  //
+  // TODO(cleanup): Should this be a public API in KJ?
```

Comment:
why not simply use `<regex>`?

Reply:
Globs are much easier to read and write than regexes, and I don't think this use case really needs the power of regexes.